### ISSUE TITLE
Training: Step 2

### ISSUE THREAD

**Issue by affromero** [facebookresearch/clevr-iep#16]
Hello,

Regarding the training procedure on step 2:
`python scripts/train_model.py  --model_type EE --program_generator_start_from data/program_generator.py --num_iterations 100000 --checkpoint_path data/execution_engine.pt`

I do not know if I have missed something, but `program_generator_start_from` is only invoked inside [get_program_generator](https://github.com/facebookresearch/clevr-iep/blob/master/scripts/train_model.py#L324-L345), for ['PG+EE' and 'PG'](https://github.com/facebookresearch/clevr-iep/blob/master/scripts/train_model.py#L171) model types. 

Thank you.

**Comment by Victoria2333** [comment]
@affromero,do you know how to compute test accuracy?coz there is no test answer in dataset,i don't know how to do it,and thanks very much~~~

**Comment by rizar** [comment]
Same question here. According to `TRAINING.md`, in Step 2 "we train the execution engine, using programs predicted from the program generator in the previous step". In `train_model.py` line 238 says "train execution engine with **ground-truth** programs". Can you please explain this discrepancy?

In any case, when I train `--model_type=EE` without Step 1 pretraining, the learning doesn't really progress (still at ~50% accuracy after 100000) iterations.

**Comment by forever010010** [comment]
have you solve the problem?

**Comment by forever010010** [comment]
But how about the val accuracy?

**Comment by rizar** [comment]
I get smth like 95-96%, which is what is reported in the paper.

**Comment by NAIZ93** [comment]
Hi,
So with 1e-5 you get 95% accuracy for execution engine? I trained it with this learning rate and also 1e-4 that was mentioned in the paper plus the default learning rate in the program, but the accuracy won't go higher than 52%.  Any suggestion?

**Comment by ankursikarwar** [comment]
@rizar can you please tell how did you get 95-96% accuracy by directly training the execution engine using the ground truth programs (as in step 2). My accuracy is oscillating around 0.47 even after 5000 iterations when using lr = 1e-5

**Comment by rizar** [comment]
Can you try training longer?

**Comment by ankursikarwar** [comment]
> Can you try training longer?

Thanks to you, I trained the execution engine for 100000 iterations with lr=1e-5 and got around 89% accuracy. Actually, accuracy increased quite slowly initially, and then between 40k and 60k iterations, it increased steeply.
![accuracy graph](https://user-images.githubusercontent.com/32908468/90718113-c6858180-e2ce-11ea-9300-98acb19b0f3f.png)

**Comment by smdp2000** [comment]
minimum pc specs you all are using to train this model, can anyone suggest
